catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java

```diff
@@ -172,6 +188,38 @@ private void updateStatus(String sourceId, QueryResponse queryResponse) {
                 State.SUCCEEDED :
                 State.FAILED));
         responseNum++;
+
+        queryResponse.getProcessingDetails()
+                .stream()
+                .filter(ProcessingDetails::hasException)
+                .forEach(details -> {
+                    status.addReason(generateSanitizedErrorMessage(details.getException(), 0));
+                });
+    }
+
+    private String generateSanitizedErrorMessage(Throwable e, int counter) {
+        if (e instanceof UnsupportedQueryException) {
+            return REASON_UNSUPPORTED_QUERY;
+        }
+
+        if (e instanceof SourceUnavailableException) {
+            return REASON_SOURCE_UNAVAILABLE;
+        }
+
+        if (e instanceof InterruptedException) {
+            return REASON_INTERRUPTED;
+        }
+
+        if (e instanceof TimeoutException) {
+            return REASON_TIMEOUT;
+        }
+
+        if (counter < MAX_EXCEPTION_SCAN_DEPTH && e.getCause() != null) {
+            counter++;
+            return generateSanitizedErrorMessage(e.getCause(), counter);
+        }
+
+        return REASON_INTERNAL;
```

Comment:
As a habit I never allow a default route of logic to lead to a recursive call. The default should end the recursion. That being said, I do like how this looks. It does appear clearer. But if something changes or breaks, and we recurse by default, the system will hang and drive us nuts down the road. Thoughts?